nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongoRecord.java

```diff
@@ -154,4 +154,28 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         }
         session.commit();
     }
+
+    private Document convertArrays(Document doc) {
+        Document retVal = new Document();
+        for (Map.Entry<String, Object> entry : doc.entrySet()) {
+            if (entry.getValue() != null && entry.getValue().getClass().isArray()) {
+                List items = new ArrayList();
+                Object[] values = (Object[])entry.getValue();
+                for (int index = 0; index < values.length; index++) {
```

Comment:
I haven't tried this yet, but does this handle arrays of arrays? If not, should it?

Reply:
I just made a change that passes validation with the original schema and that theoretically supports nested arrays. Thing is, from what I can tell you can't even express this in Avro:

```
{
    "arrayTest": [
        [[0,1], [2,3]]
    ]
}
```

Reply:
Saw mostly variations of [this](http://apache-avro.679487.n3.nabble.com/Avro-schema-for-nested-arrays-doesn-t-work-td1455659.html).